### PR TITLE
allow search hits in body text that are in title of other doc #2891

### PR DESCRIPTION
**Pull request by TimKam**

### CODE REVIEW

File: sphinx/search/__init__.py [hunk 1]
```diff
@@ -397,7 +397,9 @@ def stem(word):
             # again, stemmer must not remove words from search index
             if not _filter(stemmed_word) and _filter(word):
                 stemmed_word = word
-            if stemmed_word not in self._title_mapping and _filter(stemmed_word):
+            already_indexed = self._title_mapping.get(stemmed_word)\
+                              and docname in self._title_mapping.get(stemmed_word)
```
Comment (tk0miya) [inline sphinx/search/__init__.py]: I think `docname in self._title.mapping.get(stemmed_word, [])` is a bit simple.

File: sphinx/themes/basic/static/searchtools.js_t [hunk 2]
```diff
@@ -420,6 +420,7 @@ var Search = {
     // now check if the files don't contain excluded terms
     for (file in fileMap) {
       var valid = true;
+      //fileMap[file] = $.unique(fileMap[file]);
```
Comment (tk0miya) [inline sphinx/themes/basic/static/searchtools.js_t]: It seems this is not needed.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM wit nits

**Comment by tk0miya** [comment]
@shibukawa Do you have any opinion about this case?
In detail, please refer #2891.

**Comment by TimKam** [comment]
@tk0miya Thanks for the feedback! Worked it in.

**Review by tk0miya — APPROVED** [review]
Confirmed. Thank you for updating!

**Comment by tk0miya** [comment]
If no comments until this weekend, I will merge this.

**Comment by tk0miya** [comment]
Merged.
Thank you for contribution!

**Pull request merged**